### DISCUSSION TITLE
Does --resume work the same for yolo11 as yolo5? Do i need to adjust epochs when resuming?

### DISCUSSION THREAD

**Discussion by quitmeyer** [ultralytics/ultralytics#17342] [Q&A] [answered]
If i run a training session intended for 100 epochs, and then it stops after 20 for some reason. Then if i run the script again with the --resume flag, do i need to change the script to run for only 80 epochs? or does it rememeber it was originally trying to hit 100?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @quitmeyer, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Ultralytics Docs](https://docs.ultralytics.com) for new users where you can find many helpful resources on usage, including guidance for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples to address some common questions.

If your query is related to a potential 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to better assist in debugging the issue.

For questions related to custom training or usage scenarios, such as your question about the `--resume` flag, please provide additional details such as your training script and logs. Meanwhile, please check out our [Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) for best practices.

An Ultralytics engineer will be along shortly to provide more detailed assistance. In the meantime, consider joining our vibrant community discussions where you can connect with others on [Discord](https://ultralytics.com/discord) 🎧, [Discourse](https://community.ultralytics.com), or our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) updated. Please confirm this in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in several up-to-date verified environments with all dependencies pre-installed, including [CUDA](https://developer.nvidia.com/cuda) and [CUDNN](https://developer.nvidia.com/cudnn):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Check the status badge above to see if all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. This ensures the stability and functionality across all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

**Comment by glenn-jocher** [comment, marked as answer]
@quitmeyer when using the `--resume` flag with YOLO11, the training will automatically continue from where it left off, aiming to complete the originally intended 100 epochs without needing to adjust the epoch count.

**Reply by tirtab** [reply]
If in hyperparameter tuning, something similar might happen, how do you resume it?

**Reply by Y-T-G** [reply]
You can't resume tuning training.

**Reply by quitmeyer** [reply]
> @quitmeyer when using the `--resume` flag with YOLO11, the training will automatically continue from where it left off, aiming to complete the originally intended 100 epochs without needing to adjust the epoch count.

This helped me out again (evil windows updates decided they were more important last night than my 40+ hour training)! Thanks so much!

https://docs.ultralytics.com/modes/train/#resuming-interrupted-trainings

**Reply by glenn-jocher** [reply]
You're welcome! Glad it worked for you. Credit goes to the Ultralytics team and the YOLO community for making this feature seamless. Let us know if you have any further questions!
